### PR TITLE
Fix Publication of a scan to any target platform does not work 

### PR DESCRIPTION
**Pull request by AliceAlbano**
Issue #319
- Implement the publish for a scan when using the command hammr image publish --id --file
- Leave the behavior without the ID as it was (it does not handle scan case, it relies on appliances)

### CODE REVIEW

**Comment by Numerinico** [comment]
does this work on stable branch ? if not we should cherry pick this to the stable

File: hammr/utils/publish_builders.py [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+        printer.out("tenant in susecloud builder not found", printer.ERROR)
+        return
+    if "description" in builder:
+        pimage.credAccount.description = builder["description"]
+
+    pimage.credAccount.displayName = builder["imageName"]
+    pimage.credAccount.tenantName = builder["tenant"]
+    if "publicImage" in builder:
+        pimage.credAccount.publicImage = True if (builder["publicImage"] == "true") else False
+    # if "paraVirtualMode" in builder:
```
Comment (ludop-usharesoft) [inline hammr/utils/publish_builders.py]: As you refactor this code to a new file, now you will appear responsible of the commented code ^^.

File: tests/unit/utils/test_publish_builders.py [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+        self.assertNotEqual(pimage, None)
+        self.assertEqual(pimage.storageAccount, builder["storageAccount"])
+        self.assertEqual(pimage.container, builder["container"])
+        self.assertEqual(pimage.blob, builder["blob"])
+        self.assertEqual(pimage.displayName, builder["displayName"])
+        self.assertEqual(pimage.resourceGroup, None)
+
+    def test_publish_azure_should_return_none_when_missing_arm_container(self):
+        # given
+        builder = self.build_arm_builder("myStorageAccount", None,"myBlob", "myDisplayName", "myResourceGroup")
```
Comment (ludop-usharesoft) [inline tests/unit/utils/test_publish_builders.py]: missing space None, "myBlob"

File: tests/unit/utils/test_publish_utils.py [hunk 3]
```diff
@@ -15,351 +15,126 @@
...
-        if region is not None: builder["region"] = region
-        return builder
+        #then
+        self.assertEqual(type(published_image), type(PublishImageAws()))
 
-class TestPublishDocker(TestCase):
-    def test_publish_docker_should_return_publish_image_when_valid_entries(self):
-        # given
-        builder = self.build_builder("testNamespace", "testRepositoryName", "testTagName")
+    def test_call_publish_webservice_raises_execption_for_wrong_image_uri(self):
```
Comment (ludop-usharesoft) [inline tests/unit/utils/test_publish_utils.py]: execption -> exception

File: hammr/commands/image/image.py [hunk 4]
```diff
@@ -108,90 +109,36 @@ def help_list(self):
...
 
+        except KeyError as e:
+            printer.out("unknown error template file, key: " + str(e), printer.ERROR)
+            return 2
         except ArgumentParserError as e:
             printer.out("ERROR: In Arguments: " + str(e), printer.ERROR)
+            return 2
             self.help_publish()
+        except ValueError as e:
+            printer.out("ERROR: " + str(e), printer.ERROR)
```
Comment (Numerinico) [inline hammr/commands/image/image.py]: ERROR ERROR ? ;)

File: hammr/commands/image/image.py [hunk 5]
```diff
@@ -108,90 +109,36 @@ def help_list(self):
...
+            if do_args.id:
+                self.do_publish_with_id(do_args, template)
+            else:
+                self.do_publish_without_id(template)
 
+        except KeyError as e:
+            printer.out("unknown error template file, key: " + str(e), printer.ERROR)
+            return 2
         except ArgumentParserError as e:
             printer.out("ERROR: In Arguments: " + str(e), printer.ERROR)
```
Comment (Numerinico) [inline hammr/commands/image/image.py]: ERROR ERROR ? ;)

File: tests/unit/utils/test_publish_builders.py (hunk 2 above)
Comment (Numerinico) [reply tests/unit/utils/test_publish_builders.py]: those eyes ;)

**Pull request merged**